bindings/java/src/org/sleuthkit/datamodel/OsAccountManager.java

```diff
@@ -979,13 +979,16 @@ private void mergeOsAccountObjects(OsAccount sourceAccount, OsAccount destAccoun
 	 * Adds a rows to the tsk_os_account_attributes table for the given set of
 	 * attribute.
 	 *
+	 * This is method is synchronized to prevent multiple threads trying to 
+	 * add osAccount attributes concurrently.
+	 * 
```

Comment:
This was indeed motivated by the issue that multiple threads add osAccount attributes simultaneously and will make the osAccount object be out of sync. 

In a multi process scenario the osAccount object data update visibility issues are different. 

Since the inserts are non transactional - they will be serialized by PostgreSQL. This will guarantee data integrity

However the event carries the osAccount object list. There is a very real chance that the two objects emitted are out of sync.
CT does not rely on objects directly only its Ids and retrieve the latest when events are fired. I am not sure how they are handled in Autopsy. If you can point me to how this issue is handled elsewhere in Autopsy, I can use it as a blueprint to fixing this issue.